From 3ccded3c89de105ef8c092ddd0777547c0fb4c68 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@nowhere.net>
Date: Sun, 30 Jan 2022 11:29:33 +0100
Subject: [PATCH] CMakeLists: Add a OCPN_RELEASE option.

OCPN_CI_BUILD is used to generate a version suffix like +23.deadbee
The initial number, here 23, is by default taken from the various build
number variables set by the CI builders. OCPN_RELEASE overrides the
value obtained with a configuration value.

First usecase is Flatpak, where the version is visible for the user.

Forwarded: https://github.com/OpenCPN/OpenCPN/pull/2544
---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index cbe032536..b9f9ad0c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -321,6 +321,7 @@ option(OCPN_USE_SVG "Use SVG graphics" ON)
 option(OCPN_USE_WEBVIEW "Use wxWidget's webview addon if available" ON)
 option(OCPN_USE_LZMA "Use LZMA for chart compression" ON)
 option(OCPN_CI_BUILD "Use CI build versioning rules" OFF)
+option(OCPN_RELEASE "Hardcoded value for build number when using OCPN_CI_BUILD" OFF)
 option(OCPN_USE_SYSTEM_LIBARCHIVE "Use the libarchive version provided by the system on MacOS" ON)
 
 option(
@@ -605,6 +606,9 @@ if (OCPN_CI_BUILD)
   today(DATE)
   commit_id(COMMIT)
   build_num(BUILD_NUMBER)
+  if (NOT "${OCPN_RELEASE}" STREQUAL "")
+    set(BUILD_NUMBER ${OCPN_RELEASE})
+  endif ()
   set(VERSION_TAIL "+${COMMIT}")
   if (NOT "${BUILD_NUMBER}" STREQUAL "")
       set(VERSION_TAIL "-${BUILD_NUMBER}${VERSION_TAIL}")
-- 
2.34.1

